File: algorithm/LeetCode/Search Insert Position/Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        set<int>tmpSet(nums.begin(), nums.end());//因为set已经排好序了，所以用set
        int i = 0;
        set<int>::iterator sItor;
        if((sItor = (tmpSet.find(target))) == tmpSet.end())//不在set中的话，就先插入
            sItor = tmpSet.insert(target);
        for(auto itor = tmpSet.begin(); itor != it.first; ++itor){
            i++;   
        return i;
    }
};

File: algorithm/LeetCode/Search Insert Position/Search Insert Position.java
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int beg = 0;
        int end = nums.size() - 1;
        int mid;
        while(beg <= end){
            mid = (beg + end) >> 1;
            if(nums[mid] > target)
                end = mid - 1;
            else if(nums[mid] < target)
                beg = mid + 1;
            else
                return mid;
        }
        int sz = nums.size();
        if(end < 0) reutrn 0;    //这个地方应该注意，不要搞反了
        if(beg >= sz) reutrn sz;
        return beg;    //这一步应该注意，很关键
    }
};